pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1048,16 +1048,24 @@ def _fixture_classes_and_options(
                     'meta': _meta,
                 }
                 if 'idf' in _services:
-                    from pytest_embedded_idf import IdfSerial
-
-                    classes[fixture] = IdfSerial
-                    kwargs[fixture].update(
-                        {
-                            'app': None,
-                            'confirm_target_elf_sha256': confirm_target_elf_sha256,
-                            'erase_nvs': erase_nvs,
-                        }
-                    )
+                    from pytest_embedded_idf import IdfSerial, LinuxSerial
+
+                    if target == 'linux':
+                        classes[fixture] = LinuxSerial
+                        kwargs[fixture].update(
+                            {
+                                'app': None,
+                            }
+                        )
```

Comment:
should be the same behavior if not `esp` in _services. https://github.com/espressif/pytest-embedded/pull/175/files#diff-b5a6b0a6e20cb69a3ce927b538961a62b3a189870fb359992536e7cc2618cfa5R1033

Reply:
Should the behavior "not `esp` in `_services` have `App`" instance of the app object then (not IdfApp), right?

Reply:
no. 

```python
            if 'idf' in _services:
                if 'qemu' in _services:
                    from pytest_embedded_qemu import DEFAULT_IMAGE_FN, QemuApp
                    classes[fixture] = QemuApp
                    kwargs[fixture].update(
                        {
                            'msg_queue': msg_queue,
                            'part_tool': part_tool,
                            'qemu_image_path': qemu_image_path,
                            'skip_regenerate_image': skip_regenerate_image,
                        }
                    )
                else:
                    from pytest_embedded_idf import IdfApp
                    classes[fixture] = IdfApp
                    kwargs[fixture].update(
                        {
                            'part_tool': part_tool,
                        }
                    )
```

I think this part should be correct. Parsing `IdfApp` doesn't require `esptool`.